content/docs/aro/blob-storage/index.md

```diff
@@ -0,0 +1,242 @@
+---
+date: '2023-08-23'
+title: Azure Blob Integration
+tags: ["ARO", "Azure", "Blob"]
+authors:
+  - Daniel Penagos
+---
+
+Guide to integrate an ARO cluster with an Azure Blob in an Azure Storage Account.
+
+# Create the secret.
+
+```bash
+mkdir deploy
+tenantId=$(az account show --query tenantId -o tsv)
+
+# ---IMPORTANT
+# You need to set your location, cluster name and resource group name as environment variables
+
+# location=eastus
+# cluster_name=my-cluster
+# rg_name=myresourcegroup
+
+oc describe secret azure-cloud-provider -n kube-system
+azure_cnf_secret=$(oc get secret azure-cloud-provider -n kube-system -o jsonpath="{.data.cloud-config}" | base64 --decode)
+echo "Azure Cloud Provider config secret " $azure_cnf_secret
+
+azure_cnf_secret_length=$(echo -n $azure_cnf_secret | wc -c)
+echo "Azure Cloud Provider config secret length " $azure_cnf_secret_length
+
+aadClientId="${azure_cnf_secret:13:36}"
+echo "aadClientId " $aadClientId
+
+aadClientSecret="${azure_cnf_secret:67:$azure_cnf_secret_length}"
+echo "aadClientSecret" $aadClientSecret
+
+subId=$(az account show --query id)
+echo "subscription ID :" $subId
+
+managed_rg=$(az aro show -n $cluster_name -g $rg_name --query 'clusterProfile.resourceGroupId' -o tsv)
+echo "ARO Managed Resource Group : " $managed_rg
+
+managed_rg_name=`echo -e $managed_rg | cut -d  "/" -f5`
+echo "ARO RG Name" $managed_rg_name
+```
+
+Once we have all the environment variables set, we need to create the configuration file we will use to populate the secret.
+
+```bash
+cat <<EOF >> deploy/cloud.conf
+{
+"tenantId": "$tenantId",
+"subscriptionId": $subId,
+"resourceGroup": "$managed_rg_name",
+"location": "$location",
+"useManagedIdentityExtension": false,
+"aadClientId": "$aadClientId",
+"aadClientSecret": "$aadClientSecret"
+}
+EOF
+
+```
+
+Check all the attributes are populated. 
+```bash
+cat deploy/cloud.conf
+```
+
+Then, create the secret with the information in place.
+```bash
+export AZURE_CLOUD_SECRET=`cat deploy/cloud.conf | base64 | awk '{printf $0}'; echo`
+
+cat << EOF > deploy/azure-cloud-provider.yaml
+apiVersion: v1
+data:
+  cloud-config: ${AZURE_CLOUD_SECRET}
+kind: Secret
+metadata:
+  name: azure-cloud-provider
+  namespace: kube-system
+type: Opaque
+EOF
+
+cat deploy/azure-cloud-provider.yaml
+oc apply -f ./deploy/azure-cloud-provider.yaml
+
+```
+
+Assign permissions to the service account.
+
+```bash
+oc adm policy add-scc-to-user privileged system:serviceaccount:kube-system:csi-azureblob-node-sa
```

Comment:
any chance we can tighten up the permissions instead of privileged ?

Reply:
I have included a specific SCC for that.